backend/main/server/helpers.go

```diff
@@ -469,13 +469,67 @@ func (h *Helpers) fetchCommunity(id int) (models.Community, error) {
 	return community, nil
 }
 
-func (h *Helpers) searchCommunities(query string) ([]models.Community, error) {
-	results, err := models.SearchForCommunity(h.A.DB, query)
-	if err != nil {
-		return []models.Community{}, err
+func (h *Helpers) searchCommunities(
+	searchText string,
+	filters string,
+	pageParams shared.PageParams,
+) (
+	[]*models.Community,
+	map[string]int,
+	error,
+) {
+	if searchText == "" {
+		isSearch := true
+
+		results, _, err := models.GetDefaultCommunities(
+			h.A.DB,
+			pageParams,
+			isSearch,
+		)
+
+		if err != nil {
+			log.Error().Err(err)
+			return nil, nil, err
+		}
+
+		categories := h.categoryCountToMap(results)
+
+		return results, categories, nil
+	} else {
+		fmt.Println(filters)
+		filtersSlice := strings.Split(filters, ",")
+		results, err := models.SearchForCommunity(
+			h.A.DB,
+			searchText,
+			filtersSlice,
+		)
+		if err != nil {
+			return []*models.Community{}, nil, err
+		}
+
+		categories := h.categoryCountToMap(results)
+		return results, categories, nil
 	}
+}
 
-	return results, nil
+func (h *Helpers) categoryCountToMap(results []*models.Community) map[string]int {
+	var categoryCount = make(map[string]int)
+	for _, community := range results {
+		if community.Category != nil {
+			categoryCount[*community.Category] = *community.Category_count
+		}
+	}
+
+	return categoryCount
+}
+
+func (h *Helpers) getCategoryCount(results []*models.Community) []interface{} {
+	var resultsWithCount []interface{}
+	for _, community := range results {
+		resultsWithCount = append(resultsWithCount, community)
+	}
+
+	return resultsWithCount
```

Comment:
It looks like the count url param is not taken into account on the search result, I've created a set for testing and results array has more elements than the count number on the first pull from the backend (FE sends 10 and backend returns all results):

<img width="686" alt="Screen Shot 2022-10-06 at 11 21 34" src="https://user-images.githubusercontent.com/7526740/194338496-9bef3779-ac18-4c6f-ac9c-662b6bf813ca.png">

The other thing I noticed is that results are not filtered with the text search after the space, in this case "before 2" should return only the community that has "before 2" but it returns the same as if I would enter "before" (The name of the community makes no sense :( ). May it has to do with the configuration of the search plug-in?

Reply:
Fixed the search results count/start. Was missing from the search text query. The results are inclusive of any of the words, so because the search contains "before" it will return any community with that word, even though the search contains another search string. The search does not "narrow" with new terms. I think this makes sense for a browser page.